getMap.js

```diff
@@ -0,0 +1,137 @@
+      var protocol = req.url.split(":")[0];
+      if (protocol == "file") {
+        var path = req.url.split("://")[1];
+        fs.readFile(path, function(err, data) {
+          if (err) {
+            return callback(err);
+          }
+          var response = {};
+          response.data = data;
+          callback(null, response);
+          debug("Request for " + req.url + " complete in " + (Date.now() - start) + "ms");
+        });
+      } else {
+        request(
+          {
+            url: req.url,
+            encoding: null,
+            gzip: true
+          },
+          function(err, res, body) {
+            var duration = Date.now() - start;
+            if (duration > 500) {
+              if (res === undefined) {
+                // If request timed out response will be undefined
+                debug("Request for " + req.url + " failed in " + duration + "ms.");
+              } else {
+                // Headers are needed for debugging cases of slow responses from AWS s3
+                debug(
+                  "Request for " +
+                    req.url +
+                    " complete in " +
+                    duration +
+                    "ms.  Headers:" +
+                    JSON.stringify(res.headers || null)
+                );
+              }
+            } else {
+              debug("Request for " + req.url + " complete in " + duration + "ms");
+            }
+            if (err) {
+              callback(err);
+            } else if (res.statusCode == 200) {
+              var response = {};
+              if (res.headers.modified) {
+                response.modified = new Date(res.headers.modified);
+              }
+              if (res.headers.expires) {
+                response.expires = new Date(res.headers.expires);
+              }
+              if (res.headers.etag) {
+                response.etag = res.headers.etag;
+              }
+
+              response.data = body;
+
+              callback(null, response);
+            } else {
+              //Dont make rendering fail if a resource is missing
+              return callback(null, {});
+            }
+          }
+        );
+      }
+    }
+  });
+  return _map;
+}
+
+exports.getMap = getMap;
+exports.getMapPool = function() {
+  const factory = {
+    create: function() {
+      return new Promise(function(resolve, reject) {
+        try {
+          var map = getMap();
+          resolve(map);
+        } catch (err) {
+          console.error("Error creating map:", err);
+          reject(err);
+        }
+      });
+    },
+    destroy: function(resource) {
+      return new Promise(function(resolve) {
+        debug("Destroying map, used " + resource.useCount + " times.");
+        resource.release();
+        resolve();
+      });
+    }
+  };
+
+  const maxMapUses = 0;
+  if (maxMapUses > 0) {
+    factory["validate"] = function(resource) {
+      debug("validate");
+      return new Promise(function(resolve) {
```

Comment:
Because thats what the docs for generic-pool say to do
> validate: a function that the pool will call if it wants to validate a resource. It should accept one argument resource where resource is whatever factory.create made. Should return a Promise that resolves a boolean where true indicates the resource is still valid or false if the resource is invalid.